netconf.c

```diff
@@ -945,7 +1367,20 @@ get_query_schema (struct netconf_session *session, GNode *query, sch_node *qsche
         }
     }
 
-    get_query_to_xml (session, query, qdepth, schflags, is_subtree, xml_list);
+    return get_query_to_xml (session, rpc, query, qdepth, path, ns_href,
+                             ns_prefix, x_type, schflags, is_subtree, xml_list);
+}
+
+static void
+cleanup_on_xpath_error (struct netconf_session *session, char *attr, gchar **split,
+                        char *ns_href, char *ns_prefix)
+{
+    free (attr);
+    g_strfreev(split);
+    g_free (ns_href);
+    g_free (ns_prefix);
+    session->counters.in_bad_rpcs++;
```

Comment:
You might find that sending the RPC error also increments these counters.

Reply:
The function _send_rpc_error increments the equivalent out_ counters. The function cleanup_on_xpath_error is incrementing the in_ counters.

Reply:
OK, didn't look closely enough sorry.